analyses/ssgsea-hallmark/02-analyze-plot-ssgsea-res.R

```diff
@@ -0,0 +1,150 @@
+diseaseTypeFilt <- diseaseTypeFilt[, c("DiseaseX", "DiseaseY", colnames(diseaseTypeFilt)[1:8])]
+write.table(diseaseTypeFilt, "results/DiseaseCorrelationPathway.txt", sep="\t", row.names=F)
+
+#Now get diseases highly associated with a certain pathway
+diseaseTableTmp <- data.frame(table(diseaseTypeFilt[,c("DiseaseX", "Pathway")]));
+diseaseTableTmp <- diseaseTableTmp[diseaseTableTmp[,"Freq"]>(length(rownames(geneSetExpMat))*.25),]
+diseaseTableTmp <- diseaseTableTmp[order(-diseaseTableTmp[,"Freq"]),]
+
+#Now set up to create heatmaps
+keepPathways <- unique(diseaseTableTmp[,"Pathway"]);
+keepDisease <- unique(diseaseTableTmp[,"DiseaseX"]);
+tmpClinData <- clinData[c("short_histology", "broad_histology")];
+colnames(tmpClinData)[1] <- "Histology" 
+rownames(geneSetExpMat) <- gsub("HALLMARK", "", rownames(geneSetExpMat))
+rownames(geneSetExpMat) <- gsub("_", " ", rownames(geneSetExpMat))
+rownames(geneSetExpMat) <- trimws(rownames(geneSetExpMat))
+
+#Heatmap across all samples, only significant pathways
+if(length(keepPathways)>10)
+{
+png("plots/HeatmapPathwaysGenes_all.png", width=1080, height=720)
+pheatmap(geneSetExpMat[keepPathways,],
+border_color="black",
+color=inferno(length(geneSetExpMat) - 1), 
+annotation_col=tmpClinData[c("Histology")],
+show_colnames=F)
+dev.off()
+
+#Heatmap across significant histologies, and pathways
+tmpClinDataTmp  <- tmpClinData[tmpClinData[,"Histology"]%in%keepDisease,]
+tmpGeneSetMat <- geneSetExpMat[keepPathways,rownames(tmpClinDataTmp)]
+print(paste("The Filtered Gene Set Matrix has", nrow(tmpGeneSetMat), "rows"));
+print(paste("The Filtered Gene Set Matrix has", ncol(tmpGeneSetMat), "columns"));
+png("plots/HeatmapPathwaysGenes_Sig.png", width=1080, height=720)
+pheatmap(tmpGeneSetMat,
+border_color="black",
+color=inferno(length(geneSetExpMat) - 1), 
+annotation_col=tmpClinData[c("Histology")],
+show_colnames=F)
+dev.off()
+}
+
+if(length(keepPathways)<10)
+{
+png("plots/HeatmapPathwaysGenes_all.png", width=1080, height=720)
+pheatmap(geneSetExpMat,
+border_color="black",
+color=inferno(length(geneSetExpMat) - 1), 
+annotation_col=tmpClinData[c("Histology")],
+show_colnames=F)
+dev.off()
+
+#Heatmap across significant histologies, and pathways
+png("plots/HeatmapPathwaysGenes_Sig.png", width=1080, height=720)
+pheatmap(geneSetExpMat,
+border_color="black",
+color=inferno(length(geneSetExpMat) - 1), 
+annotation_col=tmpClinData[c("Histology")],
+show_colnames=F)
+dev.off()
+}
+
+
+
+#Function to create boxplot for certain pathways
+createBox <- function(myPathway=NULL)
+{
+	tmpDat <- cbind(tmpClinData, geneSetExpMat[myPathway,]);
+	colnames(tmpDat)[3] <- "Score"
+	ggplot(tmpDat, aes(Histology, Score))+geom_boxplot()+theme_bw()+coord_flip()+ggtitle(paste(myPathway, "- GSVA Scores"));
+	ggsave(paste("plots/", gsub("_", " ", myPathway), " GSVA_Boxplot.png", sep=""))
+
+}
+
+#Create a few boxplots to illustrate pathways that show variation among histologies
+createBox("BILE ACID METABOLISM")
```

Comment:
They are among the top significant pathways i.e. associated with particular histotypes. They are also known to be associated with proliferation and immune evasion. Bile Acid Metabolism is  sort of an outlier here but there is some literature connecting it to Glioma. Overall, my sense is that we may update or change which pathways we plot based on the rest of manuscript.

Reply:
As the data are expected to change with things being refined, it would be better to go ahead and write out all of these that meet some threshold instead of specifying things here.